Sources/LoggingTestKit/TestLogHandler.swift

```diff
@@ -0,0 +1,320 @@
+        label: String? = nil,
+        level: Logger.Level? = nil,
+        message: Regex<AnyRegexOutput>? = nil,
+        metadata: [(Logger.Metadata.Element) -> Bool] = [],
+        source: Regex<AnyRegexOutput>? = nil,
+        file: Regex<AnyRegexOutput>? = nil,
+        function: String? = nil,
+        line: UInt? = nil
+    ) -> Bool {
+        if let label,
+           self.label != label
+        {
+            return false
+        }
+        if let level,
+           self.level != level
+        {
+            return false
+        }
+        if let message,
+           !self.message.description.contains(message)
+        {
+            return false
+        }
+        for predicate in metadata {
+            if !self.metadata.contains(where: predicate) {
+                return false
+            }
+        }
+        if let source,
+           !self.source.contains(source)
+        {
+            return false
+        }
+        if let file,
+           !self.file.contains(file)
+        {
+            return false
+        }
+        if let function,
+           self.function != function
+        {
+            return false
+        }
+        if let line,
+           self.line != line
+        {
+            return false
+        }
+
+        return true
+    }
+}
+
+public struct TestLogHandler: LogHandler, Sendable {
+    public typealias Filter = (CapturedLogMessage) -> Bool
+
+    public final class Container: @unchecked Sendable {
+        // Guarded by queue:
+        private var _messages: [CapturedLogMessage] = []
+
+        private let queue: DispatchQueue
+
+        private let filter: Filter
+
+        init(_ filter: @escaping Filter) {
+            self.filter = filter
+            queue = DispatchQueue(label: "TestLogHandler.Container:\(type(of: filter))")
+        }
+
+        public var messages: [CapturedLogMessage] {
+            var result: [CapturedLogMessage] = []
+            queue.sync {
+                result = self._messages
+            }
+            return result
+        }
+
+        public func append(_ message: CapturedLogMessage) -> Bool {
+            if filter(message) {
+                queue.sync {
+                    _messages.append(message)
+                }
+                return true
+            }
+            return false
+        }
+
+        public func reset() {
+            queue.sync {
+                self._messages = []
+            }
+        }
+
+        public var factory: (String) -> TestLogHandler {
+            { label in
+                TestLogHandler(label: label, container: self)
+            }
+        }
+    }
+
+    private static let queue = DispatchQueue(label: "TestLogHandler")
+
+    // Guarded by queue:
+    private static var isInitialized = false
+    // Guarded by queue:
+    private static var logLevel: Logger.Level = .info
+    // Guarded by queue:
+    private static var _containers: [Weak<Container>] = []
+
+    public static func bootstrap() {
```

Comment:
I am leaning towards not offering this method and rather let the user in their tests do this. If they already have code that accepts a logger we just need to make the init public and they should be able to avoid bootstrapping completely.